test/components/form/error_tag_test.exs

```diff
@@ -0,0 +1,186 @@
+defmodule Surface.Components.Form.ErrorTagTest.Common do
+  @moduledoc """
+  Common functions used by both ErrorTagTest and ErrorTagSyncTest
+  """
+
+  def changeset do
+    {%{}, %{name: :string}}
+    |> Ecto.Changeset.cast(%{name: "myname"}, [:name])
+    |> Ecto.Changeset.add_error(:name, "is already taken")
+    |> Ecto.Changeset.add_error(:name, "another test error")
+    # Simulate that form submission already occurred so that error message will display
+    |> Map.put(:action, :insert)
+  end
+end
+
+defmodule Surface.Components.Form.ErrorTagTest do
+  use ExUnit.Case, async: true
+
+  import Surface
+  import ComponentTestHelper
+
+  alias Surface.Components.Form.ErrorTagTest.Common
+  alias Surface.Components.Form, warn: false
+  alias Surface.Components.Form.Field, warn: false
+  alias Surface.Components.Form.ErrorTag, warn: false
+
+  setup do
+    %{changeset: Common.changeset()}
+  end
+
+  test "multiple error messages", %{changeset: changeset} do
+    assigns = %{changeset: changeset}
+
+    code =
+      quote do
+        ~H"""
+        <Form for={{@changeset}} opts={{ as: :user }}>
+          <Field name="name">
+            <ErrorTag />
+          </Field>
+        </Form>
+        """
+      end
+
+    assert render_live(code, assigns) =~
+             "<span phx-feedback-for=\"user_name\">is already taken</span>"
+
+    assert render_live(code, assigns) =~
+             "<span phx-feedback-for=\"user_name\">another test error</span>"
+  end
+
+  test "no errors are shown if changeset.action is empty", %{changeset: changeset} do
```

Comment:
This functionality is probably more closely related to `Phoenix.HTML.Form` than `<ErrorTag>`. But I think it's worth documenting and testing so we can catch regressions.